core/src/main/java/org/apache/carbondata/core/metadata/blocklet/BlockletInfo.java

```diff
@@ -238,6 +278,20 @@ public void setNumberOfPages(int numberOfPages) {
     for (int i = 0; i < measureChunkOffsetsSize; i++) {
       measureChunksLength.add(input.readInt());
     }
-
+    // Deserialize datachunks as well for older versions like V1 and V2
```

Comment:
Can you wrap this logic into a function and mentioning it is for V1 and V2 deserialization only, I think it will be more readable

Reply:
ok